src/main/connection.cpp

```diff
@@ -71,6 +76,12 @@ std::unique_ptr<QueryResult> Connection::executeWithParams(PreparedStatement* pr
     return clientContext->executeWithParams(preparedStatement, std::move(inputParams));
 }
 
+std::unique_ptr<QueryResult> Connection::executeWithParamsWithId(
```

Comment:
Do we need this? I'm pretty sure if we prepare, we cannot support multiple statements. You can add a test case on this.